Shared/Samples/Augment reality to collect data/AugmentRealityToCollectDataView.swift

```diff
@@ -0,0 +1,210 @@
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            Divider()
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Button {
+                    treeHealthSheetIsPresented = true
+                } label: {
+                    Image(systemName: "plus")
+                        .imageScale(.large)
+                }
+                .disabled(!canAddFeature)
+                .actionSheet(isPresented: $treeHealthSheetIsPresented) {
+                    ActionSheet(
+                        title: Text("Add tree"),
+                        message: Text("How healthy is this tree?"),
+                        buttons: [
+                            .default(Text("Healthy")) {
+                                addFeature(treeHealth: .healthy)
+                            },
+                            .default(Text("Distressed")) {
+                                addFeature(treeHealth: .distressed)
+                            },
+                            .default(Text("Dead")) {
+                                addFeature(treeHealth: .dead)
+                            },
+                            .cancel()
+                        ])
+                }
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+    
+    /// Adds a feature to represent a tree to the tree survey service feature table.
+    /// - Parameter treeHealth: The health of the tree.
+    private func addFeature(treeHealth: TreeHealth) {
+        guard let featureGraphic = graphicsOverlay.graphics.first,
+              let featurePoint = featureGraphic.geometry as? Point else { return }
+        
+        statusText = "Adding feature"
+        
+        // Create attributes for the new feature.
+        let featureAttributes = [
+            "Health": treeHealth.rawValue,
+            "Height": 3.2,
+            "Diameter": 1.2
+        ] as [String: Any]
+        
+        Task {
+            if let newFeature = featureTable.makeFeature(
+                attributes: featureAttributes,
+                geometry: featurePoint
+            ) as? ArcGISFeature {
+                do {
+                    // Add the feature to the feature table.
+                    try await featureTable.add(newFeature)
+                    _ = try await featureTable.applyEdits()
+                } catch {
+                    self.error = error
+                }
+                
+                newFeature.refresh()
+            }
+        }
+        
+        statusText = "Tap to create a feature"
+        graphicsOverlay.removeAllGraphics()
+        canAddFeature = false
+    }
+}
+
+private extension AugmentRealityToCollectDataView {
+    /// The health of a tree.
+    enum TreeHealth: Int16, CaseIterable, Equatable {
```

Comment:
Conformance to `CaseIterable` isn't being used, although it probably could (and should) to create the buttons.